core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@lanwen The whole line,

```
newName != null && !newName.trim().equals(name.trim())
```

and 

```
!StringUtils.equals(trim(name), trim(newName))
```

are _not_ logically equivalent.  In the latter case, if newName is null, the body of the if-block will be entered.  In the former case, the block will _not_ be entered..